### DISCUSSION TITLE
Need Help converting .ai to .png with custom DPI Setting

### DISCUSSION THREAD

**Discussion by YakshPatel013** [ImageMagick/ImageMagick#3406] [Help]
HI:)
Context:
OS WIndows 10 64 bit
IM version: 7.0.11-Q16-HDRI
GS Version: 9.53.3
RA7.0.11-Q16-M/CPU/HD: 32/4.04GHz(8core)/1TBSSD

I am trying to convert .ai to .png with custom DPI settings at various increments  between  150-600.

I have tried the following commands and have only been able to complete the conversion of the image to a .png. However the DPI is defaulted at 96 and will not change. 

>magick input.ai -density 300 output.jpg
When i tried the same command while attempting to convert a .jpg to a .png the DPI did stick,which is why i am a bit confused. 
I also attempted this cmd
>magick convert -unit PixelsPerInch input.ai -density 300 output.png

Thank you for your assistance.

**Comment by snibgo** [comment]
You are changing the density _after_ rasterizing the input.ai file, so it just changes metadata. This should work fine. If it doesn't, please post a link to your input.ai.

But is that what you want to do? Perhaps you want to rasterize at various densities, which would create output with different numbers of pixels as well as different metadata. If that is what you want, then set "-density" _before_ reading input.ai.

**Comment by fmw42** [comment]
Put the density before reading the AI in order to get better resolution. If needed, specify other density afterwards for the PNG.
